operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/MetadataAwareOnUpdateFilter.java

```diff
@@ -0,0 +1,85 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.javaoperatorsdk.operator.processing.dependent.kubernetes;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.informers.cache.Cache;
+import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
+import io.javaoperatorsdk.operator.processing.event.source.filter.OnUpdateFilter;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * addresses the additional events noted on
+ * https://github.com/fabric8io/kubernetes-client/issues/5215
+ * <p>
+ * Several of these circumstances seem to have been addressed by ~1.25+
+ */
+public class MetadataAwareOnUpdateFilter<T extends HasMetadata> implements OnUpdateFilter<T> {
+
+  private static final Logger log = LoggerFactory.getLogger(MetadataAwareOnUpdateFilter.class);
+
+  private final OnUpdateFilter<T> underlyingFilter;
+  private final KubernetesSerialization kubernetesSerialization;
+
+  public MetadataAwareOnUpdateFilter(OnUpdateFilter<T> underlyingFilter,
+      KubernetesSerialization kubernetesSerialization) {
+    this.underlyingFilter = underlyingFilter;
+    this.kubernetesSerialization = kubernetesSerialization;
+  }
+
+  @Override
+  public boolean accept(T newResource, T oldResource) {
+    if (underlyingFilter != null && !underlyingFilter.accept(newResource, oldResource)) {
+      return false;
+    }
+
+    ObjectMeta newMetadata = newResource.getMetadata();
+    ObjectMeta oldMetadata = oldResource.getMetadata();
+    // quick check if anything meta has changed
+    if (newMetadata != null && oldMetadata != null
+        && !minimizeObjectMeta(newMetadata).equals(minimizeObjectMeta(oldMetadata))) {
+      return true;
+    }
+    // check everything else besides the metadata
+    // since the hierarchy of model may not implement hashCode/equals, we'll convert to a generic
+    // form
+    // that should be less expensive than full serialization
+    var newNode = kubernetesSerialization.convertValue(newResource, ObjectNode.class);
+    newNode.remove("metadata");
+    var oldNode = kubernetesSerialization.convertValue(oldResource, ObjectNode.class);
+    oldNode.remove("metadata");
+    if (newNode.equals(oldNode)) {
+      // could check the previous annotation to see if it's our update
+      log.warn("Ignoring revision with no actual changes for resource type {}, {}. "
```

Comment:
How often do we anticipate this situation to occur? Shouldn't this be logged at the debug level rather?